### ISSUE TITLE
Security vulnerability: fixed path in o=rwx directory in .deb

### ISSUE THREAD

**Issue by AJCxZ0** [andy-portmen/native-client#93]
The [preinst](https://github.com/andy-portmen/native-client/blob/master/apt/DEBIAN/preinst) and [postinst](https://github.com/andy-portmen/native-client/blob/master/apt/DEBIAN/postinst) files indicate the use of a fixed path in a world-writable directory and likely vulnerability to a symlink attack.
Use of mktemp(1) is strongly recommended.

I have not tested the .deb install.

No [Security policy](https://github.com/andy-portmen/native-client/security), so reporting here.

**Comment by andy-portmen** [comment]
We need a temporary place to copy the installer files during the deb generation. How about the new `/opt/native-client/`

**Comment by AJCxZ0** [comment]
No, `/opt` is not an appropriate location for temporary files and should not be writable by any user except _root_ and will therefore break all installs.
`/tmp` is the correct location, however the problem is using a known fixed path in that directory. See e.g. [CAPEC-132](https://capec.mitre.org/data/definitions/132.html). This is trivially fixed by using mktemp(1).

**Comment by andy-portmen** [comment]
We need a temporary directory that is generated during the package creation to keep the installation files. So we cannot use `mktemp`. see https://github.com/andy-portmen/native-client/blob/master/prepare.sh#L14

**Comment by andy-portmen** [comment]
@AJCxZ0 How about `/usr/share/native-client/`? 
https://github.com/andy-portmen/native-client/commit/56076588cf55c7eb151734fb2ab1ce5027da7053

**Comment by AJCxZ0** [comment]
No, `/usr/share` is also the wrong place for temporary files. The [Filesystem Hierarchy Standard](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html) describes the purpose of the various directories and local implimentation is described in _hier(7)_.
If, for reasons which are remain unclear to me, you cannot use _mktemp(1)_, then assuming derived names are similarly problematic, the old fashioned method is to start by testing for the existence of an unlikely fixed path, e.g. `/tmp/native-client.l2MdGdgVstpt7XGZKmr7JOxXmQn`, and test for its existence. If it exists and you cannot remove it, then error and exit, else create it with a suitably restricted mode such as 700, use and remove it.

The real question of how to properly build the package, but my experience of packaging on Debian is too little and too old give a useful and authoritative answer or recognise one with sufficient confidence to recommend it.

**Comment by andy-portmen** [comment]
> If, for reasons which are remain unclear to me, you cannot use mktemp(1)

We have a NodeJS executable and a few JS files that are supposed to place browser specific files inside the home directory. I need to place them inside the package and somehow have access to them to run the installer. I tried to place them inside the "DEBIAN" directory, but I couldn't find a way to have access to them during the installation. The `mktemp` works fine if you need a temp directory just during the installation.
